modules/store/src/runtime_checks.ts

```diff
@@ -69,7 +69,7 @@ export function createImmutabilityCheckMetaReducer({
 }
 
 function ignoreNgrxAction(action: Action) {
-  return action.type.startsWith('@ngrx');
+  return action.type.includes('@ngrx');
```

Comment:
Why do we need this change? I don't think it should be included in this PR

Reply:
One of the errors in #2553 was a `TypeError: Cannot add property skip, object is not extensible` thrown by the strictActionImmutability runtime check. Since action mutation is baked into ngrx/data with use of the skip and error fields in EntityActionOptions it makes sense to exempt ngrx/data actions from the runtime check. As data action types take the form: '[tag || entityname] @ngrx/data.....' they aren't captured by startsWith('@ngrx').

Reply:
AFAIK the `delete` action is the only one that's causing this problem (I could be wrong).
Can't we make it immutable instead?

Reply:
Yes this issue is only related to the delete action. But the action is mutated by design. When change tracking is enabled (by default) on entity if an add or update is persisted to the local store but not to the server (ie. no add success action is dispatched) and a subsequent delete action is dispatched the reducer removes the entity from the local store and adds the skip flag to the action so the effects ignore it and do not try to delete a record that never got to the server.

In a similar manner I believe there are cases where other actions would cause an error in a reducer method which would add the error flag to the action so that the action is ignored by the effects. From the definition of the ngrx/data entity-action: 
```javascript
...
 // Mutable actions are BAD.
  // Unfortunately, these mutations are the only way to stop @ngrx/effects
  // from processing these actions.

  /**
   * The action was determined (usually by a reducer) to be in error.
   * Downstream effects should not process but rather treat it as an error.
   */
  error?: Error;

  /**
   * Downstream effects should skip processing this action but should return
   * an innocuous Observable<Action> of success.
   */
  skip?: boolean;
}
```
I don't think we can make data actions immutable without some significant changes.

Reply:
I'd rather not change the behavior of the startsWith check, but change the action string for ngrx/data. It could be considered a breaking change is someone is listening based on that action type, but it's unlikely they are using the actual string itself.

Reply:
We could keep this in mind while working on v13 and leave it as it for now?

Reply:
Yea, i think that would be better

Reply:
Ok I can change the ngrx/data action string instead. There is a test fix and a fix for mutable state change bug also identified in #2553 in this PR as well. These would not introduce breaking changes. Should I spin them out into separate PRs so they can be merged before v13? Maybe they should have been separate from the beginning? New to the PR process.

Reply:
You can do them in PR as separate commits and we'll just keep the separate commits instead of squashing them into one